lighthouse-core/gather/gatherers/offline.js

```diff
@@ -21,14 +21,13 @@ class Offline extends Gatherer {
    * @param {LH.Gatherer.LoadData} loadData
    * @return {Promise<LH.Artifacts['Offline']>}
    */
-  afterPass(passContext, loadData) {
+  async afterPass(passContext, loadData) {
     const navigationRecord = loadData.networkRecords.filter(record => {
       return URL.equalWithExcludedFragments(record.url, passContext.url) &&
         record.fetchedViaServiceWorker;
     }).pop(); // Take the last record that matches.
 
-    return passContext.driver.goOnline(passContext)
```

Comment:
can you add a comment around L16 that the goOnline now happens in start-url which sequentially follows this gatherer?

Reply:
actually it seems like the goOffline in L16 isnt needed anymore since starturl does it on its own. right

Reply:
> can you add a comment around L16 that the goOnline now happens in start-url which sequentially follows this gatherer?

well the `goOnline` that cleans this up actually happens in `gather-runner` that was the problem. each `afterPass` happens while online and each gatherer has to handle its own online/offline cycle

> actually it seems like the goOffline in L16 isnt needed anymore since starturl does it on its own. right

it's still needed to setup the offline for the load of the page, we do the offline check for both the page and the start url separately